### PR TITLE
Feature/parameter: Add parameter dialog

### PR DESCRIPTION
**Pull request by buganini**
Requirement:
pip3 install QPUIQ==0.3

### CODE REVIEW

File: cpp/modmesh/view/RParameter.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <modmesh/view/RParameter.hpp> // Must be the first include.
+
+namespace modmesh
+{
+
+static int int64V = 5566;
```
Comment (yungyuc) [inline cpp/modmesh/view/RParameter.cpp]: I think the global static data should be managed by a singleton.  The singleton will make it possible for testing in Python (pytest).

File: cpp/modmesh/view/RParameter.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+static double doubleV = 77.88;
+
+static std::shared_ptr<pybind11::object> paramsState = nullptr;
+static std::shared_ptr<pybind11::list> paramsList = nullptr;
+
+template <typename T>
+void addParam(const char *key, T *ptr) {
```
Comment (yungyuc) [inline cpp/modmesh/view/RParameter.cpp]: Would it be possible to store the parameter information in the hierarchical dynamic toggle: https://github.com/solvcon/modmesh/blob/1bd05322720e3ce4064c7c36acccd44081c68bf1/tests/test_toggle.py#L280 ?

More information about modmesh toggles: https://github.com/solvcon/modmesh/blob/1bd05322720e3ce4064c7c36acccd44081c68bf1/cpp/modmesh/toggle/toggle.hpp#L220

**Review by yungyuc — CHANGES_REQUESTED** [review]
Thanks, @buganini .  Some items to change:

- [ ] Use C++ singleton for globals
- [ ] Use dynamic hierarchical toggles
- [ ] CI needs to be clean

Discussions:

Is it possible to have a simple unit test?

**Comment by buganini** [comment]
image

File: modmesh/params.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+from PUI.PySide6 import *
```
Comment (yungyuc) [inline modmesh/params.py]: Please add license text.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Please add copyright notice to Python files under the `modmesh` package

**Comment by buganini** [comment]
clang-format/flake8 issues are fixed

**Comment by yungyuc** [comment]
I merged the code so that we can move on.  A follow up ticket #247 is filed.

**Pull request merged**